pkg/scheduler/schedule_one.go

```diff
@@ -987,7 +987,7 @@ func (sched *Scheduler) handleSchedulingFailure(ctx context.Context, fwk framewo
 		logger.V(2).Info("Unable to schedule pod; no fit; waiting", "pod", klog.KObj(pod), "err", errMsg)
 	} else if apierrors.IsNotFound(err) {
 		logger.V(2).Info("Unable to schedule pod, possibly due to node not found; waiting", "pod", klog.KObj(pod), "err", errMsg)
-		if errStatus, ok := err.(apierrors.APIStatus); ok && errStatus.Status().Details.Kind == "node" {
+		if errStatus, ok := err.(apierrors.APIStatus); ok && (errStatus.Status().Details.Kind == "nodes" || errStatus.Status().Details.Kind == "node") { // Ususally `node` is used as the Kind, but `nodes` is also possible.
 			nodeName := errStatus.Status().Details.Name
```

Comment:
Do we need this (L989-L1001) in the first place by the way? 🤔

Reply:
Any thought? @kubernetes/sig-scheduling-approvers 

If Node is not found, the event handler should catch NodeDeleted event and remove Node from cache appropriately. I'm not sure if we have to remove Node from cache here (+ as this PR points out, this functionality hasn't been working until now, in the first place 😓 )

Reply:
I agree that this is unnecessary.
I traced the origin of this logic to here #56622 and #56261

I'm pretty sure we don't run into these problems anymore.

Reply:
@utam0k can you change the PR to remove L989-L1001 based on this discussion?

Reply:
Sure!
ad47cbac14b